.github/workflows/deployment.yml

```diff
@@ -0,0 +1,99 @@
+name: 'Deployment to GCP'
+
+on:
+  push:
+    branches:
+      - main
+
+
+jobs:
+  build-infrastructure:
+    name: 'Set up infrastructure'
+    runs-on: ubuntu-latest
+    env:
+      GOOGLE_APPLICATION_CREDENTIALS: /tmp/credentials.json
+
+    defaults:
+      run:
+        shell: bash
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set up Terraform
+        uses: hashicorp/setup-terraform@v1
+
+      - name: Set Up Credentials
+        env:
+          SET_UP_CREDENTIALS: ${{ secrets.GOOGLE_CREDENTIALS }}
+        run: printf '%s' "$SET_UP_CREDENTIALS" > "$GOOGLE_APPLICATION_CREDENTIALS"
+
+      - name: Terraform Init
+        run: |
+          cd iac/terraform
+          terraform init
+
+      - name: Terraform Plan
+        run: |
+          cd iac/terraform
+          printf "mongodburi = \"${{ secrets.MONGODB_URI }}\"" > terraform.tfvars
+          terraform plan
+
+      - name: Terraform Apply
+        if: github.event_name == 'push'
```

Comment:
is there a possibility that github.event_name == 'push' is not true?
your job only runs on push
```on:
  push:
    branches:
      - main
   ```

Reply:
maybe 🙈 
removed it